carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -369,6 +374,115 @@ void WMBroadcaster::removeGeofence(std::shared_ptr<Geofence> gf_ptr)
   carma_wm::toBinMsg(send_data, &gf_msg_revert);
   map_update_pub_(gf_msg_revert);
 };
+  
+void  WMBroadcaster::routeCallbackMessage(const cav_msgs::Route& route_msg)
+{
+
+ cav_msgs::ControlRequest cR; 
+ cR =  routeCallbackMessageLogic(route_msg);
+ route_callmsg_pub_(cR);
+
+}
+
+
+cav_msgs::ControlRequest WMBroadcaster::routeCallbackMessageLogic(const cav_msgs::Route& route_msg) 
+{
+
+
+  auto path = lanelet::ConstLanelets(); 
+
+
+  if (!current_map_) {
+// Return / log warning etc.
+  ROS_INFO_STREAM("Value 'current_map_' does not exist.");
+  throw lanelet::InvalidObjectStateError(std::string("Base lanelet map is not loaded to the WMBroadcaster"));
+
+  exit(0);
+}
+
+  for(auto id : route_msg.route_path_lanelet_ids) 
+  {
+    auto laneLayer = current_map_->laneletLayer.get(id);
+    path.push_back(laneLayer);
+  }
+ 
+  
+  if(path.size() == 0) exit(0);
+  ROS_WARN_STREAM("Got here safely");
+   /*logic to determine route bounds*/
+  std::vector<lanelet::ConstLanelet> llt; 
+  std::vector<lanelet::BoundingBox3d> pathBox; 
+  float minX = 99999;
+  float minY = 99999;
+  float minZ = 99999;
+  float maxX = -99999;
+  float maxY = -99999;
+  float maxZ = -99999;
```

Comment:
Use std::numeric_limits to get these values.
Also use doubles by default instead of floats. Stack memory is not at a premium in our system so better to prefer doubles by default.